src/lib.rs

```diff
@@ -313,6 +313,86 @@ where
     }
 }
 
+/// An iterator over the infinite range starting at `start`
+#[derive(Clone)]
+pub struct RangeFrom<A> {
+    state: A,
+    one: A,
+}
+
+/// Return an iterator over the infinite range starting at `start` and continuing forever.
+///
+/// *Note*: Currently, the `Iterator` implementation is not checked for overflow. If you use a finite-sized integer type and the integer overflows, it might panic in debug mode or wrap around in release mode. **This behavior is not guaranteed and may change at any time.**
+#[inline]
+pub fn range_from<A>(start: A) -> RangeFrom<A>
+where
+    A: Add<A, Output = A> + Clone + One,
+{
+    RangeFrom {
+        state: start,
+        one: One::one(),
+    }
+}
+
+impl<A> Iterator for RangeFrom<A>
+where
+    A: Add<A, Output = A> + Clone,
+{
+    type Item = A;
+
+    #[inline]
+    fn next(&mut self) -> Option<A> {
+        let result = self.state.clone();
+        self.state = self.state.clone() + self.one.clone();
+        Some(result)
+    }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (usize::MAX, None)
+    }
+}
+
+/// An iterator over the infinite range starting at `start` by `step`
+#[derive(Clone)]
+pub struct RangeFromStep<A> {
```

Comment:
I'm torn on the name here, compared to `RangeStepFrom`, mostly because we already have the example of `RangeInclusive` and `RangeStepInclusive`. But your way is more natural for the argument order, where `range_from_step(start, step)` means "create a range from `start` in increments of `step`."

Reply:
In terms of how each would work as a phrase, `RangeFromStep` sounds like a range from a step, but `RangeStepFrom` sounds like stepping from somewhere, which sounds more accurate. It does have the reversed argument order issue, though (and `range_step_from(step, start)` seems super confusing).

I'm open to doing it either way; let me know if I should change it.

Reply:
Also, it looks like [you did originally say](https://github.com/rust-num/num-iter/issues/16#issuecomment-615494119) `RangeStepFrom`, oops.

Reply:
That's ok, it's a reasonable evidence of intuition that you went a different way, without even realizing it.

Reply:
I guess the issue is that both are completely reasonable. I'm *slightly* inclined towards changing it, since it sounds slightly more natural, it matches existing types better, and the parameter order IMO isn't really an issue, especially if you interpret "step" as a verb instead of a noun. I can also leave as-is it if you'd prefer. Another option would be to make one a type alias for the other, but that would probably cause inconsistency and be confusing. What do you think?

Reply:
If you're still slightly inclined to change, please do, but in the end I think it won't be a big deal either way. I don't think we should bother with aliases though, just pick one and live with it.

Reply:
BTW, the function name should also follow the same order for consistency.